### PR TITLE
Fix import submodules

### PR DESCRIPTION
**Pull request by justinchuby**
### Description
Fix import submodules in the root `__init__`

### Motivation and Context

@jcwchen

### CODE REVIEW

**Review by jcwchen — APPROVED** [review]
I did see several issues asking this. (e.g., https://github.com/onnx/onnx/issues/4690) Exposing more common submodules seems fine to me.

File: onnx/__init__.py [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 # isort:skip_file
 import os
+import typing
```
[the remark this answers is not in the document]
Comment (jcwchen) [reply onnx/__init__.py:5]: Not relevant to this PR: @justinchuby do you think this check is useful? I found a lot of existing imports in ONNX break this.

[the remark this answers is not in the document]
Comment (justinchuby) [reply onnx/__init__.py:5]: I think we should disable it. lgtm's config used to include a disable on this check. I haven't figured out how to disable that on codeql but there should be a config we can set

[the remark this answers is not in the document]
Comment (jcwchen) [reply onnx/__init__.py:5]: Same thought. I will take care of it. Thanks!

[the remark this answers is not in the document]
Comment (justinchuby) [reply onnx/__init__.py:5]: lmk if you found the way and we can apply it on onnxscript as well

File: onnx/__init__.py [hunk 2]
```diff
@@ -49,38 +53,39 @@
...
-import google.protobuf.message
-
-from typing import Union, IO, Optional, cast, TypeVar, Any
+from . import (
+    checker,
+    compose,
+    defs,
+    gen_proto,
+    helper,
+    hub,
```
Comment (justinchuby) [inline onnx/__init__.py:62]: @jcwchen i didn't look to make sure the sub modules have imports that may take too long. are there modules with a long initialization time we should avoid importing here?

Comment (justinchuby) [reply onnx/__init__.py:62]: Sounds good! Feel free to merge

**Pull request merged**